File: src/Worker/content/NBB.Worker/HealthChecks/SqlConnectionHealthCheck.cs
﻿#if HealthCheck
using System.Data.Common;
using System.Data.SqlClient;

namespace NBB.Worker.HealthChecks
{
    public class SqlConnectionHealthCheck : DbConnectionHealthCheck
    {
        private static readonly string DefaultTestQuery = "Select 1";

        public SqlConnectionHealthCheck(string name, string connectionString)
            : this(name, connectionString, testQuery: DefaultTestQuery)
        {
        }

        public SqlConnectionHealthCheck(string name, string connectionString, string testQuery)
            : base(name, connectionString, testQuery ?? DefaultTestQuery)
        {
        }

        protected override DbConnection CreateConnection(string connectionString)
        {
            return new SqlConnection(connectionString);
        }
    }
}
#endif

File: src/Worker/templates/NBB.Worker.FSharp/HealthChecks/SqlConnectionHealthCheck.fs
namespace NBB.Worker.HealthChecks
#if HealthCheck

open System.Data.SqlClient
open System.Data.Common

type SqlConnectionHealthCheck(name: string, connectionString: string, ?testQuery: string) =
    inherit DbConnectionHealthCheck(name, connectionString, testQuery |> Option.defaultValue(SqlConnectionHealthCheck.DefaultTestQuery))
    
    static member DefaultTestQuery = "Select 1";
   
    override _.CreateConnection(connectionString: string) =
        new SqlConnection(connectionString) :> DbConnection

#endif

